lib/model/model.go

```diff
@@ -69,6 +69,12 @@ type Availability struct {
 	FromTemporary bool              `json:"fromTemporary"`
 }
 
+type DirectoryTree struct {
+	Name        string           `json:"name"`
+	IsDirectory bool             `json:"isDirectory"`
```

Comment:
Could use protocol.FileInfoType enum value here?
This way we don't need boolean flags for each, and you'd know about symlinks etc.

I think it should just serialise straight to a string.

Reply:
It doesn't serialise to a string 😞 
<img width="234" alt="Screen Shot 2020-01-27 at 18 18 58" src="https://user-images.githubusercontent.com/14272775/73197438-8d6aa180-4131-11ea-9aae-7270a07d9fee.png">
But I can still use `if fileType === 0 {} else if fileType === 1...` when rendering, right?

Reply:
On the need endpoints where file infos are returned, `FileInfoType.String` is used for serialization. Maybe for programmatic purposes it's even easier to deal with ints?

Reply:
Should I add an icon for `FileInfoTypeDeprecatedSymlinkFile` (2) and `FileInfoTypeDeprecatedSymlinkDirectory` (3) as well? Even if they are deprecated?

Reply:
As for me, we can have ints. It is more error-prone to compare strings than ints.

Reply:
Nono, ignore deprecated stuff for new features.

Reply:
Int's are not nice for public APIs, so let's use strings.

I suspect you won't find those two deprecated anywhere anymore, but for carefulness sake, we should just return them as symlinks.

Reply:
Okay, I'll add support for those two as well. And the thing about the public api and ints, I agree. The values should be self-explanatory for people who use just public api. I'll try to serialize them to string.

Reply:
Hmm, I see, that file which maps strings to ints is generated.